enable fp8 cast for inductor CPU

zhuhaozhe opened the pull request:
Enable FP8 cast for this issue https://github.com/pytorch/pytorch/issues/117119.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #117737



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

zhuhaozhe committed:
enable fp8 cast for inductor CPU

[ghstack-poisoned]

jgong5 commented on test/inductor/test_cpu_repro.py:
```diff
@@ -2765,6 +2771,23 @@ def forward(self, x):
+            y1 = x.to(dtype=torch.float8_e5m2).to(dtype)
+            return y0, y1
+
+        compiled_fp8_cast = torch.compile(fp8_cast, backend="inductor", dynamic=True)
+
+        shape = [int(dim) for dim in shape.split(",")]
+        x = torch.rand(*shape, device="cpu", dtype=dtype)
+        y0_fp8, y1_fp8 = compiled_fp8_cast(x)
+
+        torch.testing.assert_close(y0_fp8, x, rtol=5e-1, atol=5e-1)
+        torch.testing.assert_close(y1_fp8, x, rtol=5e-1, atol=5e-1)
```
why not just comparing with eager like other tests do?

zhuhaozhe replied:
I am copying it from the `CUDA` test before. Now I aligned it with other tests now (using `self.common`).

zhuhaozhe committed:
Update on "enable fp8 cast for inductor CPU"

Enable FP8 cast for this issue https://github.com/pytorch/pytorch/issues/117119.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

zhuhaozhe commented:
@pytorchbot rebase

zhuhaozhe committed:
Update on "enable fp8 cast for inductor CPU"

Enable FP8 cast for this issue https://github.com/pytorch/pytorch/issues/117119.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

zhuhaozhe committed:
Update on "enable fp8 cast for inductor CPU"

Enable FP8 cast for this issue https://github.com/pytorch/pytorch/issues/117119.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

zhuhaozhe commented:
@pytorchbot merge
